homeassistant/components/prometheus/__init__.py

```diff
@@ -118,10 +118,14 @@ def setup(hass: HomeAssistant, config: ConfigType) -> bool:
         default_metric,
     )
 
-    hass.bus.listen(EVENT_STATE_CHANGED, metrics.handle_state_changed)
+    hass.bus.listen(EVENT_STATE_CHANGED, metrics.handle_state_changed_event)
     hass.bus.listen(
         EVENT_ENTITY_REGISTRY_UPDATED, metrics.handle_entity_registry_updated
     )
+
+    for state in hass.states.all():
```

Comment:
Should we not apply the filter here to pass only those we care about?

Reply:
Yes, you're right. I had the filter in the handle_state function earlier, but missed it when I refactored a bit.

Reply:
Please undraft once the requested changes has been made

Reply:
I just pushed up the change, including a rebase on the current dev branch. Undrafted. Thanks for catching.